exchange/exchange.go

```diff
@@ -108,21 +117,22 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 		_, targData.cacheHost, targData.cachePath = e.cache.GetExtCacheData()
 	}
 
-	debugInfo := getDebugInfo(bidRequest, requestExt)
+	debugInfo := getDebugInfo(r.BidRequest, requestExt)
 	if debugInfo {
 		ctx = e.makeDebugContext(ctx, debugInfo)
 	}
 
 	bidAdjustmentFactors := getExtBidAdjustmentFactors(requestExt)
 
-	recordImpMetrics(bidRequest, e.me)
+	recordImpMetrics(r.BidRequest, e.me)
 
 	// Make our best guess if GDPR applies
-	usersyncIfAmbiguous := e.parseUsersyncIfAmbiguous(bidRequest)
+	usersyncIfAmbiguous := e.parseUsersyncIfAmbiguous(r.BidRequest)
 
 	// Slice of BidRequests, each a copy of the original cleaned to only contain bidder data for the named bidder
+	labels := buildLabels(r)
 	blabels := make(map[openrtb_ext.BidderName]*pbsmetrics.AdapterLabels)
-	cleanRequests, aliases, privacyLabels, errs := cleanOpenRTBRequests(ctx, bidRequest, requestExt, usersyncs, blabels, labels, e.gDPR, usersyncIfAmbiguous, e.privacyConfig, account)
```

Comment:
This is rather curious. I see we have a new line defining `labels` here, yet I do not see an old line defining `labels` that this new line is replacing. That would seem to indicate that `labels` is new to this PR, but the old line just above has it as an argument to `cleanOpenRTBRequests()`.  I must be missing something.

Reply:
Ahh, found it. It is being passed in the arguments from the endpoint, and the endpoint also builds the labels. So it seems we are tossing out that work and repeating it here. Should we just get rid of this new `buildLables()` line?

Reply:
I don't like the concept of the label structs. The code is organizing core information around the metrics system, whereas the reverse makes sense IMHO. Can't change it all overnight. We had discussed this as a first step in the metrics refactoring.

Reply:
We can move it here as well, if that is the direction we want to move. It just doesn't make sense to do the same work twice. I thought perhaps you started to move it, and then realized that it was still needed at the higher level, and forgot to remove this code when the plan fell through.

Reply:
Alright. I removed `buildLabels` and pass down the labels from the endpoints until we can factor it away entirely.